### ISSUE TITLE
OpenELEC 6.0, Wetek Play: No support for 480cvbs/i/p and  576cvbs/i/p modes in Kodi

### ISSUE THREAD

**Issue by benni0815** [OpenELEC/OpenELEC.tv#4502]
As the title says Kodi does not support the mentioned resolutions when using the analogue output and therefore switches to 720p. The problem has also been described here: http://wetekforums.com/vb5/forum/wetek-play/linux/openelec/21637-analog-av-output

I propose to cherry-pick commit xbmc/xbmc@89782b73fcf9c96698f5c083da6bcbbf11b01f83 from Kodi's Jarvis branch. I built the current 6.0 branch of OpenELEC with the patch and it worked. [Attached is a zipped patch file.](https://github.com/OpenELEC/OpenELEC.tv/files/64617/patch.zip)

Please mind that the following command has still to be executed before (re)starting Kodi:
```
echo "576cvbs" > /sys/class/display/mode
```

**Comment by codesnake** [comment]
We already used similar solution in the past, but then we had to remove it because of some problems it caused: see https://github.com/OpenELEC/OpenELEC.tv/commit/ffe39368400fb58bb59541fc015e334e1eec3ed0
We need better solution.

**Comment by benni0815** [comment]
I think that in fact I only need lines 344-379 of AMLUtils.cpp to be added. Would that be OK?

**Comment by codesnake** [comment]
If you may have noticed you refer the commit for Android, not Linux.

**Comment by benni0815** [comment]
> If you may have noticed you refer the commit for Android, not Linux.

Yes, I'm aware of that. But the changes in ```aml_mode_to_resolution``` (in xbmc/utils/AMLUtils.cpp) are relevant for the Wetek Play because this function is also used in EGLNativeTypeAmlogic.cpp. (Correct me if I'm wrong.) At least it's working on my Wetek.

> And why we should cherry-pick if it is already in Kodi repository?

Ok, sorry. I should have mentioned that I would like to backport this only to the 6.0 branch so that 6.0.x releases have it.
This commit has only been pushed to Kodi's Jarvis branch and as OpenELEC 6.0 has (and stays on?) Isengard this change is not included there yet.

**Comment by hitn** [comment]
Based on the ideas of benni0815, I had a look at the soures myself.

I propose the following patch:

[kodi-995.50-implement-disp_cap-override-fallback-and-cvbs-support.patch.zip](https://github.com/OpenELEC/OpenELEC.tv/files/69648/kodi-995.50-implement-disp_cap-override-fallback-and-cvbs-support.patch.zip)

It implements a fallback and an override mechanism for disp_cap. When there is a file /storage/.kodi/userdate/override_cap, this will override any information read out of the drivers disp_cap.

If no override_cap is present, it will read out the drivers disp_cap.

If disp_cap is empty, because there is nothing connected to the HDMI port or the HDMI display device is buggy, it tries to read /storage/.kodi/userdate/fallback_cap. If there is a fallback_cap, the capabilities found in there will be used.

With this mechanism, the user can define a fallback_cap. The values in there will only be used, when there is no HDMI display connected (or it's buggy). If the content of fallback_cap then is "576cvbs", the system will go to this mode if - and only if! - there is no HDMI display connected. It would not break any HDMI functionality.

Further the patch implements support for the modes 576cvbs and 480cvbs. Beware that this implementation differs from the one found in Jarvis since you have to destinguish between 576i and 576cvbs (resp. 480i and 480cvbs).


The drawback: It will not work out of the box, no autodetection. The user has to place a file at the appropriate location.

The advantage: Give the user (and especially for WeTek: the paying customer!!!) at least the possibility to use the features many users expects when they buy a WeTek OpenELEC box.

**Comment by tweevick** [comment]
The patch works like a charm

**Comment by tweevick** [comment]
I found it implemented by using this image:
[QUOTE=Sepp;n24465]Yes, sir! [IMG]http://wetekforums.com/vb5/core/images/smilies/icon_e_biggrin.gif[/IMG]

You can find an update image for WeTek OpenELEC 6.0.0 with CVBS analogue output support here:

[url]http://ulozto.net/xe5i15Ar/openelec-wetek-play-arm-6-0-devel-20151222130926-r21630-gba32c43-tar[/url]


Usage:
1) Copy the file to /storage/.update
2) since you're already there: go to /storage/.kodi/userdata/ and create a file "fallback_cap" with the content "576cvbs". For example, you can do this by typing: echo "576cvbs" > /storage/.kodi/userdata/fallback_cap
3) reboot your box. This will update the OpenELEC image, so give it time and don't interrupt the process!
4) Now, the box should go to 576cvbs mode when there is no HDMI monitor connected. Of course, you can't use the HDMI and the CVBS output at the same time.



Disclaimer: WARNING: This is experimental and unsupported. Use at your own risk. If you brick your box, your house burns down, your wife leaves you, well, I warned you.[/QUOTE]

**Comment by tweevick** [comment]
Important: you must have a working OpenELEC 6 Image installed. It does not work as un update for 5.x

**Comment by hitn** [comment]
So, any thoughts on this? I don't see any activity here, even though a fully functional patch was provided.

With the upcomming service release of OpenELEC on the horizon, it would be nice if this patch could be included.

**Comment by codesnake** [comment]
I think we need more seamless configuration for CVBS without a need of any additional file creation. Our previous solution was to check if HDMI is connected and if not - switch to CVBS. But this caused another problem - if someone don't want to use CVBS and always use HDMI he may get "No signal" on the TV if turned on the box before the TV.

**Comment by hitn** [comment]
Ok, I agree, that from ordinary user's point of view, this is not a perfect solution, since he has to mess around with config files.

But on the other hand: Do you have an idea to make it better - "more seamless" - at all? Autodetection is not possible, as you explained in detail, because there is no possibility to detect if something is connect to the CVBS port. This would be the best solution.

Configuration via the GUI is also not without big drawbacks, you would need an HDMI device to switch to CVBS, and vice versa.

So, what's the actual idea to make it more seamless? If you have an idea, please tell me, maybe I can support implementing that.

If you don't have an idea, then why not include my patch? It gives the user at least a chance to use this feature. It's better than domineer over the user by telling him it's too complicated and make his life either even more complicated or frustrating.

And don't forget: This mechanism does not have to stay in there forever. Once you found your better solution, implement it, drop out the lines of code of this patch and everything will be fine. But in my eyes some kind of solution which doesn't break anything else is better than no solution at all.

**Comment by tweevick** [comment]
I compiled the latest git version last week. The patch was not accepted (it did not compile). So I compiled without the patch. After reboot, cvbs output was possible after the echo command to force cvbs output. But the screen size is to big, and I cannot access the resolution settings. It should work out of the box on wetek play openelec. Why provide an analog output which does not work, I do not want to downgrade to OE 5.0.6, to watch TV, because the code is regression

**Comment by hitn** [comment]
@tweevick: Yeah, doesn't surprise me. I made the patch for 6.0.0 back than, because this was (and still is) the stable release and I expected least trouble with plugin/addon compatibility. 

Of course as development moves on, the code diverges more and more and the patch can't be applied anymore. That's why the __only__ sensible way is to merge it to the head and let it flow with the development. Otherwise, you would have to "fork" it again and again with every change in the head branch.

But you see here in the discussion how "supportive" and "interested" the code maintainers are. You know, I'm not a OpenELEC or Kodi developer. I'm just a user who bought quite a bunch of these WeTeK boxes and was then very disappointed, that an advertised feature did not work. I completely agree with you: It should work out of the box!!! But it does not. 
But then again I thought it's not about whining and demanding, it's open source and also as a user I'm part of the community. So I did my share, developed a solution and offered help. More I can't do.

I'll probably create another patch with next stable release, if I think it's worth updating my boxes to the next release. After all it's not rocket science. But I can't promise you this.

**Comment by tweevick** [comment]
I contacted wetek and this was their answer:
 Helpdesk (WeTek)

Jan 27, 11:44

Mr. Steveninck,

Although we're sponsors of OpenELEC, it is still an independent project with open community of users and developers, and where particular implementations could also have impacts on all other OpenELEC deployments. As far as we could get, A/V ouput (audio is still working) was creating conflicts with HDMI ouput. In case you install official Android ROM, you'll get A/V output port working.

Thanks for your suggestions and inputs.
Regards,
Paulo

**Comment by rgenoud** [comment]
I just bought a wetek openelec because it had a video analog output...

**Comment by tweevick** [comment]
Read `http://www.wetekforums.com/v/index.php?p=/discussion/21637/analog-av-output/p1`

<blockquote class="Quote" rel="Sepp">Since WeTek is still doing false advertising and not bothering about a non-funtional AV output at all, the community still has to take care about this on its own.

So, here I provide you an updated OpenELEC 6.0.3 build with the above described features, terms and conditions, which will let you use your AV output:

http://ulozto.net/xJ2zpGAC/openelec-wetek-play-arm-6-0-3-cvbs-tar


If you would like to see your hardware work as advertised with stock firmware and not only with poorly maintained community builds (namely mine ;-), go to Github issue tracker and join the bug report:

https://github.com/OpenELEC/OpenELEC.tv/issues/4502
</blockquote>

**Comment by tweevick** [comment]
You can buy a cable at https://wetek.com/shop/av-cable-wetek-play?lang=fra

**Comment by rgenoud** [comment]
Thanks for your links, I ordered a cable yesterday, and downloaded your 6.0.3-cvbs build.
Have you a github of the sources you used for this build ?

**Comment by tweevick** [comment]
Contact sepp on the wetek forum. I just provided the links.

**Comment by hitn** [comment]
@rgenoud: No, there is no github of the sources. There is only the ZIP file I posted above which contains the patch.

It still applies well against 6.0.3. You can just check out the normal git sources with the appropriate tag, include my patch following the development workflow found here

http://wiki.openelec.tv/index.php/Development_workflow

and build your own built.

Full ACK about your comment with the "preferably without an out-of-tree patch" part. Maybe it's time to move on to LibreELEC? Maybe they care about this issue.

**Comment by rgenoud** [comment]
2016-04-14 22:21 GMT+02:00 hitn <notifications@github.com>:
>
> @rgenoud: No, there is no github of the sources. There is only the ZIP file I posted above which contains the patch.
>
> It still applies well against 6.0.3. You can just check out the normal git sources with the appropriate tag, include my patch following the development workflow found here
>
> http://wiki.openelec.tv/index.php/Development_workflow
>
> and build your own built.
>
> Full ACK about your comment with the "preferably without an out-of-tree patch" part. Maybe it's time to move on to LibreELEC? Maybe they care about this issue.

Ok, thanks.
I build my own openelec-6.0.3cvbs, it works as expected !

**Comment by tweevick** [comment]
Can you explain how to patch the openelec source to enable cvbs.

**Comment by rgenoud** [comment]
2016-04-24 20:15 GMT+02:00 tweevick <notifications@github.com>:
> Can you explain how to patch the openelec source to enable cvbs.

Yes, I've just forked openelec on github and add instructions in the README
https://github.com/rgenoud/OpenELEC.tv/tree/openelec-6.0-cvbs

I'll may add another repository with binary build.

**Comment by zwobbl** [comment]
I've build your fork a few days ago and it works perfectly for me.
I would like to see the patch integrated upstream ...

**Comment by rgenoud** [comment]
For those who are interested, I've made a github with the binaries for {Open,Libre}ELEC with CVBS support.
https://github.com/rgenoud/libreELEC-CVBS-binaries

**Comment by tweevick** [comment]
Great, 
Important: the binaries created are  tar.xz and won't update, you need to convert them to tar files, otherwise the update does not work.

Once in *.tar converted and and put on the wetek:
scp target/LibreELEC-WeTek_Play.arm-7.0.2cvbs.**tar** wetek_ip_address:/storage/.update/
ssh wetek_ip_address "echo 576cvbs > /storage/.kodi/userdata/fallback_cap"

Thanks a lot

**Comment by hitn** [comment]
With the commit of https://github.com/xbmc/xbmc/pull/11159/commits/9353b0cdec3869bbfd85347a6ba53e32dfdef511 (and maybe others), this issue should be solved now. It's supported now in LibreELEC 9 and Kodi 18. 

(Eventhough we could have had this 3 years earlier :roll_eyes:)
